bankdroid-legacy/src/main/java/com/liato/bankdroid/banking/banks/AbsIkanoPartner.java

```diff
@@ -125,8 +125,8 @@ public Urllib login() throws LoginException, BankException, IOException {
     @Override
     public void update() throws BankException, LoginException, BankChoiceException, IOException {
         super.update();
-        if (username == null || password == null || username.length() == 0
-                || password.length() == 0) {
+        if (getUsername() == null || getPassword() == null || getUsername().length() == 0
```

Comment:
That could be a good idea. I think an empty string by default should do it. then we don't need to check for null values.